Remotion/Web/Core/Context/HttpContextStorageProvider.cs

```diff
@@ -34,6 +34,11 @@ public class HttpContextStorageProvider : ISafeContextStorageProvider
 #pragma warning restore 618
 #endif
 
+    public SafeContextBoundary OpenSafeContextBoundary ()
+    {
+      return default;
```

Comment:
I think we need the new semantic with the cleanup also for HttpContextStorageProvider in .NET 5. 

And since the semantics shouldn't differ between .net 5 and .net Framework, we also need this for Callcontext-Storageprovider.

Reply:
`HttpContextStorageProvider` writes the values directly into `HttpContext.Items`. How should a reset behave in this case?

Reply:
hmm....removing and re-adding while the scope is open. this will "guard" scenarios such as parallel.foreach. For task etc, httpcontext should not propagate to the new thread anyway.

Reply:
So we need to keep track of all values written to be able to remove those values at the boundary and restore the old values afterwards? But what about values from the inside scope leaking out?

We would need to save the complete `Items` collection and set it back the way it was. Which is kind of awkward since we cannot set the `Items` directly, but have to make a diff and do `Add`/`Remove` calls

Reply:
Actually: place a dictionary into the Items and switch the dictionary. Same as with AsyncLocal

Reply:
(I know, it's different than before and should be documented, but over all, that solves things.)

Reply:
Performance question: 
1. the lookup speed will depend on the key-length, the key should be just a few characters (see SafeContextKeys).
2. how does the performance of a weakreference-valueget compare to the lookup in the HttpContext.Items? If the weakreference is significantly better, we could discuss a performance optimization by caching the last dictonary in a threadlocal weakreference (there's a bit more involved...)